bootstrap/eksctl/permission-boundary.json

```diff
@@ -31,39 +31,6 @@
           }
         }
       },
-      {
-        "Sid": "DenyAccessIfRequiredPermBoundaryIsNotBeingApplied",
-        "Effect": "Deny",
-        "Action": [
-          "iam:PutUserPermissionsBoundary",
-          "iam:PutRolePermissionsBoundary"
-        ],
-        "Resource": [
-          "arn:aws:iam::ACCOUNT_ID:user/*",
-          "arn:aws:iam::ACCOUNT_ID:role/*"
-        ],
-        "Condition": {
-          "StringNotEquals": {
-            "iam:PermissionsBoundary": "arn:aws:iam::ACCOUNT_ID:policy/crossplaneBoundary"
-          }
-        }
-      },
-      {
-        "Sid": "DenyRoleCreationWithOutPermBoundary",
-        "Effect": "Deny",
-        "Action": [
-          "iam:CreateRole"
-        ],
-        "Resource": [
-          "arn:aws:iam::ACCOUNT_ID:user/*",
-          "arn:aws:iam::ACCOUNT_ID:role/*"
-        ],
-        "Condition": {
-          "StringNotEquals": {
-            "iam:PermissionsBoundary": "arn:aws:iam::ACCOUNT_ID:policy/crossplaneBoundary"
-          }
-        }
-      },
```

Comment:
Any reason for removing these?

Reply:
The policy as written indicates that the iam:CreateRole action will only be deny when the role does not have a permission boundary set. However, in contrast when ArgoCD tried to create the role in the s3-irsa example, it was denied permission. Removal of this policy entry allowed ArgoCD to create the role.

Reply:
I think that means the s3-irsa is not properly handling permission boundary. Permissions boundary is not implemented in the terraform version for some reason so that's probably why.